nextcord/ui/select/channel.py

```diff
@@ -0,0 +1,271 @@
+        channels: List[Union[GuildChannel, Thread]] = []
+        guild_channels = None
+        for id in self.data:
+            channel = guild.get_channel_or_thread(id)
+            if channel is None:
+                if guild_channels is None:
+                    guild_channels = await guild.fetch_channels()
+                channel = get(guild_channels, id=id)
+                if channel:
+                    channels.append(channel)
+            else:
+                channels.append(channel)
+        return channels
+
+
+class ChannelSelect(SelectBase):
+
+    """Represents a UI channel select menu.
+
+    This is usually represented as a drop down menu.
+
+    In order to get the selected items that the user has chosen,
+    use :attr:`ChannelSelect.values`., :meth:`ChannelSelect.get_channels`
+    or :meth:`ChannelSelect.fetch_channels`.
+
+    .. versionadded:: 2.3
+
+    Parameters
+    ----------
+    custom_id: :class:`str`
+        The ID of the select menu that gets received during an interaction.
+        If not given then one is generated for you.
+    placeholder: Optional[:class:`str`]
+        The placeholder text that is shown if nothing is selected, if any.
+    min_values: :class:`int`
+        The minimum number of items that must be chosen for this select menu.
+        Defaults to 1 and must be between 1 and 25.
+    max_values: :class:`int`
+        The maximum number of items that must be chosen for this select menu.
+        Defaults to 1 and must be between 1 and 25.
+    disabled: :class:`bool`
+        Whether the select is disabled or not. Defaults to ``False``.
+    row: Optional[:class:`int`]
+        The relative row this select menu belongs to. A Discord component can only have 5
+        rows. By default, items are arranged automatically into those 5 rows. If you'd
+        like to control the relative positioning of the row then passing an index is advised.
+        For example, row=1 will show up before row=2. Defaults to ``None``, which is automatic
+        ordering. The row number must be between 0 and 4 (i.e. zero indexed).
+    channel_types: List[:class:`nextcord.ChannelType`]
+        The types of channels that can be selected. If not given, all channel types are allowed.
+    """
+
+    __item_repr_attributes__: Tuple[str, ...] = (
+        "placeholder",
+        "min_values",
+        "max_values",
+        "disabled",
+        "channel_types",
+    )
+
+    def __init__(
+        self,
+        *,
+        custom_id: str = MISSING,
+        placeholder: Optional[str] = None,
+        min_values: int = 1,
+        max_values: int = 1,
+        disabled: bool = False,
+        row: Optional[int] = None,
+        channel_types: List[ChannelType] = MISSING,
+    ) -> None:
+        Item.__init__(self)
+        self._selected_values: List[str] = []
+        self._provided_custom_id = custom_id is not MISSING
+        custom_id = os.urandom(16).hex() if custom_id is MISSING else custom_id
```

Comment:
Why is this code copied and pasted from `SelectBase.__init__`? Can't you just use `super().__init__()`?